### PR TITLE
fix(ec): prefer credible replica as canonical metric in EC detection

### PR DESCRIPTION
**Pull request by chrislusf**
## Problem

An interrupted EC encode can leave a 0-byte `.dat` replica behind (a stub). When that stub sits on a lower-sorting server than the real replica, detection's lowest-server canonical pick reported `Size=0`, tripped the min-size gate, and the volume was stranded in `skippedTooSmall`. Detection never proposed an encode, so the partial EC shards from the failed attempt were never cleared, and every re-distribute kept hitting the `ec volume %d is mounted; refusing overwrite` guard.

This is the cross-server slice of the stuck-encode reports on #9478: the existing same-store prune (`pruneIncompleteEcWithSiblingDat`) won't act because the only sibling `.dat` is the 0-byte stub, and the worker's `cleanupStaleEcShards` never runs because detection isn't firing.

## Change

`selectCanonicalMetric` now picks the lowest-server **credible** replica — data-bearing (`.dat` larger than a superblock) and not already EC — instead of the unconditional lowest server. Consequences:

- A 0-byte stub on a low server no longer becomes canonical, so the volume's real replica drives the size/fullness checks and the encode proceeds. On success the originals (including the stub) are deleted and the partial EC is cleared, unsticking the volume.
- A leftover EC shard set on a low server no longer short-circuits at the `IsECVolume` guard, so the orphan-source cleanup (#9448) and re-encode paths get their chance.
- When nothing is credible there is nothing to encode, so it falls back to the lowest-server metric and the downstream gates skip exactly as before.

The decision lives in a small pure helper (`planning.go`) so it is unit-testable without a live cluster.

## Tests

- `isStubReplica` boundary cases around the superblock floor.
- `selectCanonicalMetric`: prefers credible over a low-server stub; skips EC metrics; falls back to lowest server when all stubs; tie-breaks by server among credibles; empty input.
- `TestDetectionEncodesDespiteLowServerStub`: end-to-end regression — a volume with a low-server stub and a higher-server real replica is proposed for EC instead of dropped as too-small.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved erasure-coding detection and source selection to ignore invalid/minimal replicas and skip already-encoded volumes, ensuring correct proposals for encoding even when the lowest-server replica is a stub.

* **Testing**
  * Added unit and regression tests covering stub detection, canonical-source selection (including ties, skips, fallbacks, and empty cases) and a regression validating encoding is still proposed when the lowest server holds a stub.

[Review Change Stack](https://app.coderabbit.ai/change-stack/seaweedfs/seaweedfs/pull/9717?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: weed/worker/tasks/erasure_coding/planning.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/seaweedfs/seaweedfs/weed/worker/types"
+)
+
+// isStubReplica reports whether a regular replica's .dat is too small to hold
+// any data — at most a superblock. An interrupted encode can leave a 0-byte
+// .dat behind; such a stub is not an encode source and must not be counted
+// toward size/fullness checks.
+func isStubReplica(size uint64) bool {
+	return size < uint64(super_block.SuperBlockSize)
+}
```
[the remark this answers is not in the document]
Comment (chrislusf) [reply weed/worker/tasks/erasure_coding/planning.go:15]: Good catch — applied in aa5c4adce. The strict `<` also contradicted the doc comment ("at most a superblock"). Now `<=` so an 8-byte data-less .dat can't shadow a real replica.

File: weed/worker/tasks/erasure_coding/planning_test.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIsStubReplica(t *testing.T) {
+	assert.True(t, isStubReplica(0), "0-byte .dat is a stub")
+	assert.True(t, isStubReplica(uint64(super_block.SuperBlockSize)-1), "below a superblock is a stub")
+	assert.False(t, isStubReplica(uint64(super_block.SuperBlockSize)), "a bare superblock is an empty-but-real volume, not a stub")
+	assert.False(t, isStubReplica(200*1024*1024), "a data-bearing replica is not a stub")
+}
```
[the remark this answers is not in the document]
Comment (chrislusf) [reply weed/worker/tasks/erasure_coding/planning_test.go:24]: Done in aa5c4adce — the bare-superblock case now asserts true, plus a SuperBlockSize+1 case for the credible boundary.
